### ISSUE TITLE
SNT versions >4.0.3 not available with pyimagej

### ISSUE THREAD

**Issue by carshadi** [morphonets/SNT#137]
Using the latest SNT or any version down to `4.0.4`
```python
ij = imagej.init(["sc.fiji:fiji", "org.morphonets:SNT"])
```

Yields
```shell
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In [3], line 1
----> 1 ij = imagej.init(["sc.fiji:fiji", "org.morphonets:SNT:4.0.4"])
      2 Tree = jimport("sc.fiji.snt.Tree")
      3 TreeAnalyzer = jimport("sc.fiji.snt.analysis.TreeAnalyzer")

File /opt/conda/envs/myenv/lib/python3.10/site-packages/imagej/__init__.py:1496, in init(ij_dir_or_version_or_endpoint, mode, add_legacy, headless)
   1493     raise EnvironmentError("Sorry, the interactive mode is not available on macOS.")
   1495 if not sj.jvm_started():
-> 1496     success = _create_jvm(ij_dir_or_version_or_endpoint, mode, add_legacy)
   1497     if not success:
   1498         raise RuntimeError("Failed to create a JVM with the requested environment.")

File /opt/conda/envs/myenv/lib/python3.10/site-packages/imagej/__init__.py:1701, in _create_jvm(ij_dir_or_version_or_endpoint, mode, add_legacy)
   1698 sj.config.endpoints.extend(original_endpoints)
   1700 try:
-> 1701     sj.start_jvm()
   1702 except subprocess.CalledProcessError as e:
   1703     # Check to see if initialization failed due to "un-managed" imagej-legacy
   1704     err_lines = []

File /opt/conda/envs/myenv/lib/python3.10/site-packages/scyjava/_java.py:177, in start_jvm(options)
    175     endpoints = endpoints[:1] + sorted(endpoints[1:])
    176     _logger.debug("Using endpoints %s", endpoints)
--> 177     _, workspace = jgo.resolve_dependencies(
    178         "+".join(endpoints),
    179         m2_repo=scyjava.config.get_m2_repo(),
    180         cache_dir=scyjava.config.get_cache_dir(),
    181         manage_dependencies=scyjava.config.get_manage_deps(),
    182         repositories=repositories,
    183         verbose=scyjava.config.get_verbose(),
    184         shortcuts=scyjava.config.get_shortcuts(),
    185     )
    186     jpype.addClassPath(os.path.join(workspace, "*"))
    188 # HACK: Try to set JAVA_HOME if it isn't already.

File /opt/conda/envs/myenv/lib/python3.10/site-packages/jgo/jgo.py:701, in resolve_dependencies(endpoint_string, cache_dir, m2_repo, link_type, update_cache, force_update, manage_dependencies, repositories, shortcuts, verbose)
    698 relevant_jars.append(jar_file_in_workspace)
    700 try:
--> 701     link(
    702         os.path.join(m2_repo, *g.split("."), a, version, jar_file),
    703         jar_file_in_workspace,
    704         link_type=link_type,
    705     )
    706 except FileExistsError as e:
    707     # Do not throw exceptionif target file exists.
    708     pass

File /opt/conda/envs/myenv/lib/python3.10/site-packages/jgo/jgo.py:215, in link(source, link_name, link_type)
    213 except OSError as e:
    214     if e.errno != 18:
--> 215         raise e
    216 try:
    217     return link(source=source, link_name=link_name, link_type="soft")

File /opt/conda/envs/myenv/lib/python3.10/site-packages/jgo/jgo.py:212, in link(source, link_name, link_type)
    210 elif link_type.lower() == "auto":
    211     try:
--> 212         return link(source=source, link_name=link_name, link_type="hard")
    213     except OSError as e:
    214         if e.errno != 18:

File /opt/conda/envs/myenv/lib/python3.10/site-packages/jgo/jgo.py:207, in link(source, link_name, link_type)
    205     return os.symlink(source, link_name)
    206 elif link_type.lower() == "hard":
--> 207     return os.link(source, link_name)
    208 elif link_type.lower() == "copy":
    209     return shutil.copyfile(source, link_name)

FileNotFoundError: [Errno 2] No such file or directory: '/home/jupyter/.m2/repository/org/jzy3d/jzy3d-native-jogl-core/2.0.0/jzy3d-native-jogl-core-2.0.0-tests.test-jar' -> '/home/jupyter/.jgo/sc.fiji/fiji/RELEASE/70b963d29e7b64a35d3d3527a5a4f0d10e6e49d601efc595497d04abd98bb31d/jzy3d-native-jogl-core-2.0.0-tests.test-jar'
```

**Comment by tferr** [comment]
I feel this happened before... and you fixed it by explicitly declaring on the pom what is needed and what is not!? Did we forget to merge that work?  I will look.

To be clear, it is all caused by not finding this jar: `/.m2/repository/org/jzy3d/jzy3d-native-jogl-core/2.0.0/jzy3d-native-jogl-core-2.0.0-tests.test-jar`, correct? (I cannot test it right now, but if that is the case, placing a dummy jar there would maybe work!? That jar is only needed at compile time.  Also SNT 4.1.11 uses jzy3d 2.0.2. but I am not sure pyimagej sees it already.

**Comment by carshadi** [comment]
@tferr putting a dummy file in that location, in addition to `/home/jupyter/.m2/repository/org/jzy3d/jzy3d-core/2.2.0/jzy3d-core-2.2.0-tests.test-jar` fixes it.

```python
SNTUtils = jimport("sc.fiji.snt.SNTUtils")
print(SNTUtils.VERSION)
```
prints `4.1.11`

**Comment by tferr** [comment]
@carshadi, I knew I had seen this before!! This is the same issue as described in https://github.com/imagej/pyimagej/issues/222. The discussion is a bit long, but the relevant part [starts here](https://github.com/imagej/pyimagej/issues/222#issuecomment-1257192428). Curtis found the culprit and fixed this in jgo back in September. Can you check if his workaround (cloning jgo locally) works? I guess the issue exists because the latest [jgo in PyPI is still from May](https://pypi.org/project/jgo/#history ). Not so sure what we should do in the interim

**Comment by carshadi** [comment]
yep, cloning `jgo` also works. I suppose I can wait for the PyPI release.

**Comment by tferr** [comment]
Curtis just released jgo 1.0.5 (https://github.com/scijava/jgo/issues/85) so everything should work without having to clone jgo
